server/aesdsocket.c

```diff
@@ -180,90 +215,135 @@ void thread_to_send(void *threadparam)
-                                func_close();
                                 free(read_buf);
       				 free(rec_buf);
-
+				close(fd);
+                                func_close();
+                                
                         }   
                 
+                syslog(LOG_USER ," write nr = %d", nr);
                 
                  if (sigprocmask(SIG_UNBLOCK,&threadlocal->mask,NULL) == -1) {   
                perror("unblock failed");
                }
                 
                 
                 
-                syslog(LOG_USER, " received rec_buf = %s", rec_buf);
+                
         
                 
 		pthread_mutex_unlock(&mutex);
         
          
           
         
-        // lseek(fd, 0, SEEK_SET);
+ 
          
-         int sent=0;
-         //while(sent< total_buffer)
-        // {
          
-                //lseek(fd, sent, SEEK_SET);
-                
-                //int read_len;
-               // if((total_buffer-sent)<buf_size) read_len=total_buffer-sent;
-                //else read_len=buf_size;
-                
+ 	int i=0, j=0;
+ 	int sent_bytes= buf_size;
                 
                 pthread_mutex_lock(&mutex);
                 
                  if (sigprocmask(SIG_BLOCK,&threadlocal->mask,NULL) == -1) {   
                perror("block failed");
                }
                 
-                int offset= lseek(fd, 0, SEEK_END);
+              
           
-        	 read_buf= (char *)realloc(read_buf, sizeof(char)*(offset));
+        	 /*read_buf= (char *)realloc(read_buf, (total_buffer+1));
+        	 if(read_buf == NULL)
+               		{
+               		
+               			perror("Malloc unsuccessful\n");
+              					free(read_buf);
+      						free(rec_buf);
+               	 			func_close(); 
+                				
+               		
+               		}*/
         	 
-        	 lseek(fd, 0, SEEK_SET);
+        	char one_byte;
+        	
+        	 lseek(fd,0,SEEK_SET);
+        	
+        	
+               while((len3 = read(fd, &one_byte, sizeof(char)))>0)
+               {
                
-         
-                len3= read(fd, read_buf, offset);
-                if(len3 == -1)
-                {
-                        perror("Read unsuccessful\n");
-                       func_close();   
-                       free(read_buf);
-         		free(rec_buf);
-        
-                }
-                
-                
-                
-                sent+=len3;   
-                
-                //stat("/var/tmp/aesdsocketdata.txt", &st);
-                
-                
-        
-        ret5= send(threadlocal->accept_fd, read_buf, len3, 0);
-         if(ret5 == -1)
-         {
-                perror("Send unsuccessful\n");
-                func_close(); 
-                free(read_buf);
-          	free(rec_buf);
-         
-          }
+               	read_buf[i]= one_byte;
```

Comment:
@chaisuresh I see the error at https://github.com/cu-ecen-aeld/assignment-8-chaisuresh/runs/4041413614?check_suite_focus=true#step:6:555.  I'm not sure what went wrong with the refactor here based on code review, but what I'd suggest is to add some logging about what you are receiving and what you are sending.  Assuming you can't reproduce in your native build I'd look closely at the differences in logging between native build and running on QEMU.